server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -700,31 +571,7 @@ public long getTimePartition() {
    * @throws PartitionViolationException if the data of the file cross partitions or it is empty
    */
   public long getTimePartitionWithCheck() throws PartitionViolationException {
-    long partitionId = -1;
-    for (Long startTime : startTimes) {
-      long p = StorageEngine.getTimePartition(startTime);
-      if (partitionId == -1) {
-        partitionId = p;
-      } else {
-        if (partitionId != p) {
-          throw new PartitionViolationException(this);
-        }
-      }
-    }
-    for (Long endTime : endTimes) {
-      long p = StorageEngine.getTimePartition(endTime);
-      if (partitionId == -1) {
-        partitionId = p;
-      } else {
-        if (partitionId != p) {
-          throw new PartitionViolationException(this);
-        }
-      }
-    }
-    if (partitionId == -1) {
-      throw new PartitionViolationException(this);
-    }
-    return partitionId;
+    return timeIndex.getTimePartitionWithCheck(file.toString());
```

Comment:
You use `getAbsolutePath()` previously but `toString()` here, will this make any difference?

Reply:
The codes used to be like this...  Actually here the `file.toString()` is just used for exception information,
while the `file.getAbsolutePath()` is used for path split.